packages/flutter/test/cupertino/route_test.dart

```diff
@@ -937,6 +938,150 @@ void main() {
     );
   });
 
+  group('Cupertino page transitions', () {
+    CupertinoPageRoute<void> buildRoute({required bool fullscreenDialog}) {
+      return CupertinoPageRoute<void>(
+        fullscreenDialog: fullscreenDialog,
+        builder: (_) => const SizedBox(),
+      );
+    }
+
+    testWidgets('when route is not fullScreenDialog, it has a barrierColor', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: SizedBox.expand(),
+        ),
+      );
+
+      tester.state<NavigatorState>(find.byType(Navigator)).push(
+        buildRoute(fullscreenDialog: false),
+      );
+      await tester.pumpAndSettle();
+
+      expect(tester.widget<ModalBarrier>(find.byType(ModalBarrier).last).color, const Color(0x18000000));
+    });
+
+    testWidgets('when route is a fullScreenDialog, it has no barrierColor', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: SizedBox.expand(),
+        ),
+      );
+
+      tester.state<NavigatorState>(find.byType(Navigator)).push(
+        buildRoute(fullscreenDialog: true),
+      );
+      await tester.pumpAndSettle();
+
+      expect(tester.widget<ModalBarrier>(find.byType(ModalBarrier).last).color, isNull);
+    });
+
+    testWidgets('when route is not fullscreenDialog, it has a _CupertinoEdgeShadowDecoration', (WidgetTester tester) async {
```

Comment:
This (and the other test below) seem to always pass. For example, you could change all colors to `0xffffffff`, and it still passes. The reason for that is that `something` must throw in invalid matches, when it returns `false`, it just skips the canvas call, so nothing actually is being verified.

I would ultimately expect this test to fail on master.

Reply:
Oh ok so I thought about this a lot, and after reading through the documentation, I ended up coming to the conclusion that the implementations for those test helpers were incorrect, and I ended up writing this to address that:
- The issue: https://github.com/flutter/flutter/issues/95981
- The PR: https://github.com/flutter/flutter/pull/95537

However, as you'll see in each of those, I understood that it's possible that I was just misunderstanding how `something` and `everything` are supposed to work—is that something you can speak to?

> The reason for that is that something (and other paints matchers) must throw in invalid matches

Specifically, my understanding was, that the `something` matcher wouldn't throw on the first invalid match, because it should keep evaluating calls until it has determined that there are no valid matches at all. Is that an incorrect reading of the docs?

https://github.com/flutter/flutter/blob/a0f8937b7c6eea33b6916e5b1dd4a1b591d9d227/packages/flutter/test/rendering/mock_canvas.dart#L396-L417

Reply:
I suppose either way, even if that is the case, then this test should be written in a way that doesn't assume that the other PR will get merged, so I'll update these tests to be meaningful in the absence of those other changes 👍🏻

Reply:
Ah, so that's actually a bug, I see. I'm not super familliar with `paints`, but what you describe in #95981 seems correct to me. I think you should probably wait until someone can verify your other PR and then we can verify this test is indeed correct.

Reply:
> I suppose either way, even if that is the case, then this test should be written in a way that doesn't assume that the other PR will get merged, so I'll update these tests to be meaningful in the absence of those other changes 👍🏻

If you can do that, it would be cool!

Reply:
Done. Here's the format of the new output when I change one of the calls to expect an incorrect color:

![image](https://user-images.githubusercontent.com/7915388/147954077-d5ccd0b8-afb8-46ca-9b07-cfaba77da11c.png)